src/spec/create_extension_spec.py

```diff
@@ -0,0 +1,531 @@
+                doc="Shape of the contact; e.g. 'circle'",
+                dtype="text",
+                quantity="?",
+            ),
+            NWBDatasetSpec(
+                name="radius_in_um",
+                neurodata_type_inc="VectorData",
+                doc="Radius of a circular contact, in micrometers.",
+                dtype="float",
+                quantity="?",
+            ),
+            NWBDatasetSpec(
+                name="width_in_um",
+                neurodata_type_inc="VectorData",
+                doc="Width of a rectangular or square contact, in micrometers.",
+                dtype="float",
+                quantity="?",
+            ),
+            NWBDatasetSpec(
+                name="height_in_um",
+                neurodata_type_inc="VectorData",
+                doc="Height of a rectangular contact, in micrometers.",
+                dtype="float",
+                quantity="?",
+            ),
+        ],
+    )
+
+    probe = NWBGroupSpec(
+        neurodata_type_def="Probe",
+        neurodata_type_inc="Device",
+        doc="Specific instance of a neural probe object.",
+        links=[
+            NWBLinkSpec(
+                name="probe_model",
+                doc="The model of the probe used to record the data.",
+                target_type="ProbeModel",
+            ),
+        ],
+        attributes=[
+            NWBAttributeSpec(
+                name="identifier",
+                doc="Identifier of the probe, usually the serial number.",
+                dtype="text",
+                required=False,
+            ),
+        ],
+    )
+
+    probe_model = NWBGroupSpec(
+        neurodata_type_def="ProbeModel",
+        neurodata_type_inc="Device",
+        doc=(
+            "Neural probe object, compatible with the ProbeInterface specification. The name of the object should "
+            'be the model name of the probe, e.g., "Neuropixels 1.0".'
+        ),
+        groups=[
+            NWBGroupSpec(
+                name="contacts_table",
+                neurodata_type_inc="ContactsTable",
+                doc="Neural probe contacts, compatible with the ProbeInterface specification",
+            ),
+        ],
+        attributes=[
+            # inherits name, description, manufacturer from Device
+            NWBAttributeSpec(name="ndim", doc="dimension of the probe", dtype="int", default_value=2),
+            NWBAttributeSpec(
+                # although the ProbeModel also has a name attribute, the name must be unique across all
+                # devices in the NWB file, and users may decide to use a more descriptive name than just
+                # the model name
+                name="model",
+                doc='Name of the model of the probe, e.g., "Neuropixels 1.0".',
+                dtype="text",
+            ),
+            NWBAttributeSpec(
+                name="planar_contour_in_um",  # TODO should this just be "contour"?
```

Comment:
Yeah, I think planar also is kind of redundant. I think boundary_countour is better.

For reference this is the way that open cv uses contours is the just the list of vertices that define a boundary but I think matplotlib has another meaning. 

Not sure here, maybe @alejoe91 can sheed some ligth on why they decide to emphasize the planar part.